src/Managers/DataManager.py

```diff
@@ -16,17 +16,32 @@ def __init__(self, data: Data, model: Model) -> None:
         """Initialize the internal data object."""
         self._data = data
         self._model = model
+        """row_begin, row_end, columns begin, columns end"""
+        self._range = [0, 0, 0, 0]
+
 
     def upload_file(self, file_path: Any) -> None:
         """Read file to the pandas format and store it."""
         self._data.file = pd.read_csv(
             filepath_or_buffer=file_path, header=0, skipinitialspace=True
         )
         self._data.columns = list(self._data.file.columns)
+    def set_range_grid(self, begin_r: Protocol, end_r: Protocol,
+                       begin_c: Protocol, end_c: Protocol) -> None:
```

Comment:
I'm not sure why do you use `Protocol`s here? I assume these values are just `int`s.